lib/compilers/rust.ts

```diff
@@ -338,4 +338,9 @@ export class RustCompiler extends BaseCompiler {
         );
         return super.runCompiler(compiler, newOptions, inputFilename, execOptions);
     }
+
+    override isOutputLikelyLlvmIr(options: string[]): boolean {
+        const emitIndex = options.indexOf('--emit');
+        return options.includes('--emit=llvm-ir') || (emitIndex >= 0 && options[emitIndex + 1] == 'llvm-ir');
```

Comment:
won't this trigger an out of bounds error if you do `--emit` without anything after it?

Reply:
OOB index access should just return `undefined`, but I can add some tests.